### DISCUSSION TITLE
Running Different Server Types/Modded (Forge, Fabric, etc)

### DISCUSSION THREAD

**Discussion by BigCrazyOfficial** [thecoder-001/MineColab#34] [Q&A] [answered]
My friends and I wanted to use this server as a modded server, since we don't quite have the money to afford a server host at the moment. This may be a stupid question, but how would I go about installing forge, or any other server for that matter? Thanks!

-big crazy

**Comment by thecoder-001** [comment, marked as answer]
Hello!
Well, if you are new to Minecraft servers the easiest way would be setup a server on your computer to your liking (you can refer to the usual "how to setup a Minecraft server" guides) , grab its folder and then dump its contents to the `Minecraft-server` folder on your google drive. Then change the server jar name in the google colab repo to whatever its in your folder.
Basically:
 1. Setup a server (forge/fabric) on your pc locally.
 2. Go to the server folder (where it stores all it stuff) and copy all the stuff within.
 3. Paste it in the `Minecraft-server` folder (make one if it doesn't already exists) in your drive.
 4. Now either change the sever.jar name by reaming it to `server.jar` **or** modify the `-jar server.jar` (change server.jar to whatever your server jar file is named) flag in the following line in the first cell of colab notebook.
  ```
!java -Xmx6144M -Xms6144M -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=https://mcflags.emc.gs -Daikars.new.flags=true -jar server.jar nogui
```
Report back if it works out for you or if you need more help.
Cheers!

**Reply by BigCrazyOfficial** [reply]
Thanks! I'll try that out as soon as I get the chance, and let you know how it goes.

**Reply by BigCrazyOfficial** [reply]
Alright, thanks! It worked, though I had a bit of trouble joining after installing the mods, I managed to figure it out. Thanks for the help!

**Reply by eggfry1** [reply]
> Hello!
> Well, if you are new to Minecraft servers the easiest way would be setup a server on your computer to your liking (you can refer to the usual "how to setup a Minecraft server" guides) , grab its folder and then dump its contents to the `Minecraft-server` folder on your google drive. Then change the server jar name in the google colab repo to whatever its in your folder.
> Basically:
> 
> 1. Setup a server (forge/fabric) on your pc locally.
> 2. Go to the server folder (where it stores all it stuff) and copy all the stuff within.
> 3. Paste it in the `Minecraft-server` folder (make one if it doesn't already exists) in your drive.
> 4. Now either change the sever.jar name by reaming it to `server.jar` **or** modify the `-jar server.jar` (change server.jar to whatever your server jar file is named) flag in the following line in the first cell of colab notebook.
> 
> ```
> !java -Xmx6144M -Xms6144M -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=https://mcflags.emc.gs -Daikars.new.flags=true -jar server.jar nogui
> ```
> 
> Report back if it works out for you or if you need more help.
> Cheers!

I am a bit confused. I would like to use colab to host a 1.12.2 forge server with my friends but don't understand how to do that based on these instructions. I created a local server with the correct world and mods we will be using, and then I copied all of those files into "Minecraft-server". I then started the server and it appears the server is still using paper. Do you have any more in-depth explaination on how to do this.

Thank you in advance,
eggfry1

**Reply by turtleship69** [reply]
Is forge's server called `server.jar`?

**Reply by BigCrazyOfficial** [reply]
Not exactly who I expected to see in my inbox today, but let's roll with it:

When you first set up the server, set `1.12.2` as the version of paper you want to be installed.

If that doesn't work, ensure **every** file within the "Minecraft-server" folder is deleted before replacing the files with the server you've created on your own PC. 

I'd also recommend setting up a new directory for every server you set up: for example, my test server used "Minecraft-server", but when I originally set up my server, I forgot to change it, meaning that both of my servers just used the same directory, achieving nothing.

Also, as the above comment mentioned, ensure that your `server.jar` is in fact titled `server.jar`, as that might cause some issues.

Hope this helps!

-big crazy
